feeds/npm/npm.go

```diff
@@ -102,18 +102,35 @@ func New(eventHandler *events.Handler) *Feed {
 
 func (feed Feed) Latest(cutoff time.Time) ([]*feeds.Package, error) {
 	pkgs := []*feeds.Package{}
+	packageChannel := make(chan *feeds.Package)
+	errs := make(chan error)
+
 	packages, err := fetchPackages()
 	if err != nil {
 		return pkgs, err
 	}
+
 	for _, pkg := range packages {
-		v, err := fetchVersionInformation(pkg.Title)
-		if err != nil {
+		go func(pkg *Package) {
+			v, err := fetchVersionInformation(pkg.Title)
+			if err != nil {
+				errs <- err
+				return
+			}
+			feedPkg := feeds.NewPackage(pkg.CreatedDate.Time, pkg.Title, v, FeedName)
+			packageChannel <- feedPkg
+		}(pkg)
+	}
+
+	for i := 0; i < len(packages); i++ {
+		select {
+		case pkg := <-packageChannel:
+			pkgs = append(pkgs, pkg)
+		case err := <-errs:
 			return pkgs, fmt.Errorf("error in fetching version information: %w", err)
 		}
-		pkg := feeds.NewPackage(pkg.CreatedDate.Time, pkg.Title, v, FeedName)
-		pkgs = append(pkgs, pkg)
 	}
+
 	feed.lossyFeedAlerter.ProcessPackages(FeedName, pkgs)
```

Comment:
How does the NPM API work for the version information? Is there chance of overlap detection for lossy feeds?

Reply:
The package specific endpoint only provides the `latest` version, so without known previous version state or enforcement of versioning patterns then I think it's worth doing the overlap check.

Reply:
Yep, looks like there's a fundamental problem with the package logic here with npm parsing. Will be tackled via #106